### DISCUSSION TITLE
还需要stylelint-config-prettier吗？

### DISCUSSION THREAD

**Discussion by sun-shu** [umijs/umi#12041] [General]
stylelint-config-prettier
Note
As of Stylelint v15 [all style-related rules have been deprecated](https://stylelint.io/migration-guide/to-15#deprecated-stylistic-rules). If you are using v15 or higher and are not making use of these deprecated rules, [this plugin is no longer necessary](https://stylelint.io/migration-guide/to-15#:~:text=Additionally%2C%20you%20may%20no%20longer%20need%20to%20extend%20Prettier%27s%20Stylelint%20config).

<img width="1095" alt="image" src="https://github.com/umijs/umi/assets/45038559/70325604-4b5e-41ec-b626-3245149c82f5">

<img width="1010" alt="image" src="https://github.com/umijs/umi/assets/45038559/eec1f72a-45ba-4278-9bb1-6b1d68ec050f">
umi lint使用的是新版的stylelint, 这里stylelint-config-prettier还是旧的版本，是否不需要这个插件了？

**Comment by fz6m** [comment]
不需要了，`@umijs/lint` 这里面的这个包做的时间也比较早，主要用于蚂蚁内部，现在伴随着 stylelint 的废弃与不在流行，以及 eslint flat config 、rust 等新 lint 工具的出现，未来可能会以 rust 的 biome 或者 oxc 为主流，如果要格式化的话建议还是用 biome 不要用这个 `@umijs/lint` 了。

另外除了组件库以外格式化 css 的意义不大，css 的新语法也是出现的非常快的，lint 基本跟不上（除非你一直升级，这是不现实的），同时 css in js 和现在 stylx 等 precompiled atom 的 css in js 也非常流行，是不可能 cover 这么多场景的 css lint 的，所以不建议使用任何形式的 css lint 了（或者就使用 tailwind + 相关 prettier 插件来排序）。
